### ISSUE TITLE
sorted_imports in 0.62.0 doesn't correctly distinguish groups

### ISSUE THREAD

**Issue by jszumski** [realm/SwiftLint#6317]
### Bug Description

The rewritten `sorted_imports` rule doesn't correctly distinguish import groups that also contain comments.  Leading comments stay with the import node they're associated with, but I'd expect that to break on newlines.

Bad correction I get when I expected the original code not to trigger a violation:

```diff
+// comment for second group
+import AAA
 //
 // Copyright © Foo
 //
 
 import DDD
 import FFF
-
-// comment for second group
-import AAA
 import BBB
 import CCC
```

Related example where the correction is placed above the first item's trivia when I would expect it to split on the new line:

```diff
+import AAA
 //
 //  Copyright © Foo
 //
 
 import class BBB.View
-import AAA
```

### Environment

* SwiftLint 0.62.1
* Xcode 26.0.1
* installed from source

**Comment by SimplyDanny** [comment]
Nobody would believe me, but the hardest part of every rule implementation always is correct comment and whitespace handling ...

**Comment by jszumski** [comment]
It's even harder since there isn't a whitespace/trivia only AST node.  Thanks for the quick turn around!
